### PR TITLE
Style changes to custom components gallery

### PR DESCRIPTION
**Pull request by aliabd**
Made some changes, but open to feedback. The gallery feels a bit wordy to me but I think the descriptions are kind of nice to read. Maybe down the line we can add some screenshots or something. 

I didn't change the actual content in the card too much because I think we're going to replace it with @pngwn's auto docs

### CODE REVIEW

**Comment by abidlabs** [comment]
Nice @aliabd! I really like it.

To reiterate some of my feedback from sync:
* I think we should remove the tags
* Would be great to use the HF API to get the custom components (any Space with tag `gradio-custom-component`) as well as the number of likes, and to display the number of likes (the components should be sorted based on # of likes as well)
* 

Some additional feedback:
image
* This header is nice, but the link is incorrect
* As in the Guides, it would be nice if the search bar showed the total number of components as well:

image

**Comment by aliabd** [comment]
thanks @abidlabs! other call feedback (just listing for completeness) 

- replace the content of the card with just the space, which will contain the auto docs 
- remove the name (or at least "gradio_") as its not as useful as the description 
- remove the tags but keep them as searchable.

**Comment by abidlabs** [comment]
Perhaps we should also include what base component a custom component was templated from. This could be helpful for users. I believe that we include this as a tag when a custom component is being created. If not, we should!

**Comment by aliabd** [comment]
Made a lot of changes: 


gallery


- Removed tags 
- Showing number of likes
- Sorting by likes 
- Fixed header 
- Showing number of components in search bar 
- Replaced card with just the space 
- Removed “gradio_” from the titles 
- Showing the templates
- Better hovering
- Showing the author

I didn't switch to using hf api because that's actually what @freddyaboulton is already doing in https://huggingface.co/spaces/gradio/custom-component-gallery-backend/blob/main/app.py exactly like we want. I think it was just easier in python and the space made sense as a backend. I think it's still curated though based on this comment https://github.com/gradio-app/gradio/pull/6477#issue-2000129693

Also currently I'm getting likes in a very dumb slow way because i couldn't figure out how to add it to freddy's space. Probably because webhooks aren't set up? Will need his help with that. But for now results will lag and resort themselves..

Will also refactor a lot of the code and move it to page.server.ts

**Comment by abidlabs** [comment]
Amazing UX @aliabd (love the hover)! Just a few little things from my end:

* In the case that the template is Fallback, I think we should just remove it. That means that the user didn't specify a template when creating the custom component
* The default font size for the description feels a bit large to me, and results in many of the descriptions being cut off at the default zoom in my browser (though the ellipses are a nice UI):

image

* Yeah we need to fix that lag/resorting...

But overall super nice! Excited to release this

File: js/_website/src/routes/custom-components/gallery/+page.svelte [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+		)
+			.then((response) => response.json())
+			.catch((error) => `Error: ${error}`);
+		define_colors(components);
+		if (!components_length) {
+			components_length = components.length;
+		}
+		for (const component of components) {
+			let info;
+			info = await fetch(`https://hf.co/api/spaces/${component.id}`)
```
Comment (freddyaboulton) [inline js/_website/src/routes/custom-components/gallery/+page.svelte:63]: The backend returns likes now so we can get rid of this

**Review by freddyaboulton — APPROVED** [review]
Great pr @aliabd !! I love the new style.

All of the feedback from @abidlabs has been addressed and I just tested and works great. Combining with the forthcoming ParamViewer is going to be 🔥 

One thing, not everyone will update their components and use the ParamViewer. I wonder if as a fallback we can at least link to the space so that users can look at the code? Thanks!

**Comment by aliabd** [comment]
Sounds good @freddyaboulton I added a link to the space. 

I removed my likes code so now everything is snappy again :) I also now load all the spaces in the beginning and clicking on one just toggles hidden. So now clicking around is much faster.

**Comment by freddyaboulton** [comment]
> I also now load all the spaces in the beginning and clicking on one just toggles hidden.

I'm a little wary about this. That's what we do in the custom components gallery and people have complained that it impacts the performance of the page. Since we load the iframe "in the background" I think in this case it may be ok. Let's give it a shot since the page is very smooth right now.

Thanks @aliabd ! Good to merge from my point of view. Love the go to space button!

**Comment by aliabd** [comment]
Got it will keep this in mind and if there are performance issues we can definitely go back to the original way. 

we should probably wait for autodocs before merging as merging this will launch it on the website

**Comment by freddyaboulton** [comment]
> we should probably wait for autodocs before merging as merging this will launch it on the website

Autodocs is an opt-in feature. After it's released the existing custom components won't update to use it automatically. So I think it's good to release without it?

**Comment by abidlabs** [comment]
Before we make the launch announcement, can we update some of the most popular custom components to include autodocs

**Comment by pngwn** [comment]
Autodocs will be done shortly so we can release that today and get some spaces using it before making noise on twitter.
